### DISCUSSION TITLE
Faculdade realmente é necessário?

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#1184] [Discussão]
Enfim, desde o ano passado comecei ir atrás de faculdade  porém as duas faculdades que iniciei eu tive que trancar por falta dinheiro. Já cheguei a ver algumas issues antes de abrir essa, mas gostaria de ouvir outras respostas/experiências. Tenho vontade de morar  fora e estou correndo atrás da documentação.   A falta de faculdade dificulta em alguma coisa? Ou vale a pena eu ir atrás de técnico? Pelo menos, para ter o diploma na área.

**Comment by romulo2735** [comment]
Cara assim, a curva para o sucesso sem ter diploma é mais "longa" a meu vê. Eu estou fazendo ADS, mas fazendo no meu tempo, sem pressa, as vagas que conseguir foi estudando por fora com cursos. Mas a faculdade abre muito a mente para conhecimentos que você não tem idea. 

Exemplo...
Sou programador PHP, aprendi MySQL e tals, mas nem sonhava que teria e havia Modelagem de Dados, aprendi e vi mais afunda na faculdade. 

Acho que o cursos + faculdade são complementares, cursos ajudam, mas a estrada é mais complicada, porque aprender sozinho, para algumas coisa limita, eu gosto de ter o professor do lado, ouvindo e tals.}

Essa é minha opnião!

**Comment by felipefialho** [comment]
Linkando: #1101 #809

**Comment by ghost** [comment]
@LFeh  vi essas ambas issues mas ainda sim não achei o suficiente para tirar minhas duvidas sobre.

**Comment by ghost** [comment]
Por mais chato que seja abrir outra issue semelhante(caso tenha problema, me desculpe)  resolvi abrir pra ouvir outras experiências além dessas.

**Comment by felipeorlando** [comment]
@brunojokubauskas sua meta é ir pra qual país?

 Do pouco que conheço, nos EUA e Canadá a maioria pede diploma ou um certo tempo de experiência (superior a 5 anos).

Digo por experiência própria, a Europa é BEM mais simples, com exceção a grandes empresas e empresas mais conservadoras.

**Comment by andersoonweb** [comment]
Vou dar minha opinião livre; Não tenho experiência alguma e nunca procurei trabalho no exterior, mas já ouvir falar que precisa comprovar formação. Vai também da região que você quer ir.

Sobre faculdade cara, acho que depende muito de você. Nessa área você consegue se igualar a qualquer um que fez algum curso superior estudando em casa, mas o maior desafio é se organizar e não procrastinar. Veja, na faculdade você tem datas e prazo para poder cumprir, caso não bata certinho, você pode perder na disciplina. Estou falando de estudos e projetos para obter notas.
Sozinho é um pouco diferente, há datas e prazos também com seus estudos mas se tu quiser deixar pra próximo mês é BEM mais fácil, isso pode te atrapalhar de monte. As vezes você pode, porém, pensa que não pode, consegue compreende? Melhor, você pode dormir mais tarde e acordar cedo, estudar mais um pouco, codar mais um pouco..

Não indico faculdade, a não ser se você tiver um objetivo e esse objetivo dependa exclusivamente dela. Mas pra conhecimento acho que autodidata é o caminho. Existe cursos perfeitos e acessível - udemy tá ai.
Você pode estudar "de graça" ou c investimento de 50 reais mês, não precisa pagar o absurdo de uma faculdade, visto que, esse é o seu problema.
Se for por esse caminho, sugiro utilizar Google Sheets, principalmente a tabela - tipo o excel, para poder se organizar.
Algo tipo isso:
![image](https://user-images.githubusercontent.com/3930770/44036026-35632252-9ee7-11e8-9d80-3dcbcdd3cb4a.png)
![image](https://user-images.githubusercontent.com/3930770/44036036-3bd5f772-9ee7-11e8-9e70-252819c8af40.png)

 [Veja](https://medium.freecodecamp.org/how-i-got-a-second-degree-and-earned-5-developer-certifications-in-just-one-year-while-working-and-2b902ee291ab), vai entender mais do que estou falando.

PS: Se quiser estudar c custo zero de investimento em materiais, indico Freecodecamp, MDN e praticar. 
Youtube, artigos e documentações vai te deixar num nível bem legal.
Ajudar também o outro é bem legal e você aprende muito. Estou fora da área mas não deixo eu sair dela por completo, estou sempre buscando ajudar o próximo. Obviamente tenho que fazer algumas pesquisar para tal.
Espero ter ajudado!

**Comment by felipeorlando** [comment]
Reforço o que o @andersoonweb disse.

Tenho exp de 4 anos profissional na área, mas estudo programação em geral a mais tempo que isso por conta própria. Só comecei a fazer faculdade a uma semana atrás e o que mais ouvi do pessoal que trabalha comigo (e que inclusive tem boas formações; USP, Poli...) foi: POR QUE?

**Comment by andersoonweb** [comment]
Pois bem @felipeorlando, algumas áreas já nos permite ter essa liberdade. Acredito que, Design, Eng da Computação - Para ser programador, Jornalismo, arte visuais, tende a ir pelo autodidata. A maior vantagem que vejo é que não há limites e maior aproveitamento de tempo.

Veja, hoje faço Letras na **UFRB**, vou há faculdade quatro dias das semanas quando não há aulas de reposições aos sábado. Cada dia em média eu fico 6/7 horas na faculdade e transporte, enfim. Multiplicando por 4 dá 24/28 horas semanais, fora o que vai ser estudado no fim de semana e madrugas. Converta isso em autodidata com foco, vai ser MUITO mais produtivo.

Na faculdade você vai ver bastante coisa legais, networking, vai possibilitar estágio dentre outras coisas. Mas fora dela, você que dita seu ritmo. Talvez você estude um semestre em dois semestre por conta própria. E o melhor, você anda de acordo com a necessidade e o mercado.

_É só um exemplo, porém acredito que pra lecionar na área de Letras realmente é preciso esse processo._
**Bom, esse posicionamento é válido pra quem quer ser PROGRAMADOR.**

**Comment by thismarcoantonio** [comment]
@andersoonweb massa ver esse post do Beau, via muito conteudo dele antigamente, bem interessante.

Sobre faculdade, acho que o importante é não achar que faculdade vai ser tudo em sua vida. Respondendo sua pergunta, não, faculdade não é realmente necessária.

**Comment by ghost** [comment]
Então, a faculdade é um pouco complicado  pois tenho uma certa pressão familiar por conta de família, principalmente minha mãe. Por mais que eu fale que na área que quero seguir faculdade não é necessário, ela ainda insiste. Porém, invés de investir numa faculdade, estou com um planejamento de colocar meu passaporte em processo e ir pra Irlanda como estudante e ficar trabalhando/estudando inglês por lá até meu passaporte ficar pronto(no caso 1 ano e meio, segundo o consulado).  Na Irlanda tenho a facilidade de renovar o visto de estudante 2x e dura 8 meses com direito a trabalho(praticamente o tempo do passaporte).  Sobre udemy, vou adquirir dois cursos que me chamou bastante atenção que é esse: https://www.udemy.com/web-completo/ do Jorge Sant Ana (achei bem interessante a abordagem do curso com vários projetos).  E vou comprar esse do Stephen Grider(gostei bastante da abordagem dele) https://www.udemy.com/coding-interview-bootcamp-algorithms-and-data-structure/ 
Meu único problema seria realmente pressão familiar por conta da faculdade.

**Comment by ViniciusGularte** [comment]
Poxa @brunojokubauskas , também sofri muito pressão familiar, meus pais queriam que eu fizesse engenharia da computação... não liga pra isso, muitas vezes nossos pais só querem o melhor pra nós, mas eles não tem a nossa mesma visão  da área  que trabalhamos; no brasil, ainda se tem a visão arcaica que precisa de um diploma pra ser alguém ... um lugar que recomendo muito além da udemy é o [FreeCodeCamp](http://freecodecamp.org/) totalmente grátis, você vai completando os desafios e ganha certificado por isso, além de principalmente praticar o que está aprendendo e criar projetos que vão contar ao seu portfólio.

Complementando.. veja o que vale o investimento para você, um diploma facilita a entrada em alguns países, mas acredito que tudo tem a hora certa entende, eu também tenho vontade de morar fora, estou juntando pra isso :smile: , boa sorte na jornada que escolher.

**Comment by juliovedovatto** [comment]
Excelente tema! Vamos usar ele em roda de discussão no FEMUG Maringá 15 😃

**Comment by thiagosantoscunha** [comment]
A 4 anos atrás resolvi fazer a minha faculdade de Engenharia da Computação. Precisava de uma profissão para garantir o meu futuro. E sim! Foi bem na época do Boom da economia brasileira e havia acabado de ser demitido de uma empresa onde ate hoje não recebi os meus direitos legais.
**Por que estudar para a área de TI se não é uma profissão regulamentada?**
Bem... a área de TI, especificamente na área de programação me trouxe grandes oportunidades para cuidar de mim e da minha família. É claro, faculdade de Eng. da Computação não é só programação, tem matemática, Eletrônica, robótica, tem redes, tem até religião e ética. Eu, vejo todas essas matérias em comum no dia a dia de um profissional. Por exemplo: Não consigo ver a programação sem as matérias de cálculo. Ela me ajudou a pensar com mais clareza, afinal, programação existe graças a matemática.

É lógico, você não precisa de uma faculdade para trabalhar e ganhar um bom salário. Mas um certificado é uma forma de provar que vc é capaz. A faculdade certifica que você é alguém.
Eu nem terminei a minha faculdade, e já recebi ótimas propostas nacionais e no exterior. Poderia ter conseguido sem a faculdade? Eu, particularmente, acredito que não!

**Comment by tcelestino** [comment]
Depende do que você realmente quer. A faculdade vai te trazer bastante retorno teórico se os professores forem atualizados e a faculdade ter um curso bastante "pra frente". Se você for bom, certamente as empresas vão querer você fazendo parte da equipe, claro que existem empresas (as grandes corporações) que para ocupar uma vaga você precisa ter um papel certificando que você está gabaritado para a mesma.

**Comment by rubenmarcus** [comment]
Se voce pesquisar no Linkedin, as melhores vagas no Brasil (falo de beneficios e salários), exigem algum diploma ( digo empresas grandes ) ,  mas o mercado é muito grande e tem pessoas que trabalham e ganham bem sem diploma. Tudo vai depender das suas soft skills também.

**Comment by kelciocajueiro** [comment]
Eu fiz e não me arrependo.

**Comment by rubenmarcus** [comment]
Estava dando uma olhada nos Degrees de faculdades gringas, e vai de 15mil dolares a 25mil dolares, e duração de 3 anos , no caso da de 15mil dolares / 36 = 470 dolares aproximadamente isso dá uns 1800 reais por mes (contando o dolar a 4 reais)

1800 reais é um valor que dá pra parcelar no cartão de crédito ou pagar caso seja até 30% do seu salario e vc nao tenha muitas contas.

Mas Vale a pena?

Eu leio muito o 'Quora', e li que nos EUA  a maioria dos estudantes estão endividados e muitos em subempregos.

Vale a pena gastar quase 50mil numa graduação online, ?? , digo temos muitos exemplos de pessoas que foram pra fora (EUA,Europa, Dubai, Asia) e não tem graduação , fora que 50mil é a entrada num ape aqui.

Aqui algumas:
Arizona State University : 15mil dolares - de 18 a 36 meses
https://www.coursera.org/degrees/master-of-computer-science-asu

Illinois University : 21 mil dolares -12 a 36 meses
https://www.coursera.org/degrees/master-of-computer-science-illinois

Master of Computer and Information Technology - 20 - 40 meses
https://www.coursera.org/degrees/mcit-penn   - 25mil dolares


Cursos de graduação online no Brasil , só vi Anhanguera e Estacio, e essas, sinceramente, acho que ninguem olha direito em processos de seleção

**Comment by rubenmarcus** [comment]
Esse da Fiap alguem recomenda?
https://www.fiap.com.br/online/graduacao/bacharelado/sistemas-de-informacao/

**Comment by simaodeveloper** [comment]
Essa aqui sai pelo menos $4000 o Bacharel em Ciências da Computação https://www.uopeople.edu/

**Comment by adhenrique** [comment]
@rubenmarcus eu recomendo. Faço ADS na Fiap, e posso dizer - na minha humilde opinião - que nunca tive professores de DDD e BD tão bons quanto os que conheci na Fiap. Há outros professores de outras áreas com conteúdo muuuuuito conceitual e massante, mas em contrapartida, muito bem estruturado.

**Comment by adhenrique** [comment]
Respondendo a dúvida do @brunojokubauskas ...

Vou compartilhar rapidamente um pouco da minha história. Trabalho com PHP e Javascript há mais ou menos uns 10 anos. Sempre me aprofundei em videos, leitura, cursos, etc... pois _achava_ que faculdade seria uma perca de tempo.

Em contrapartida, todo tempo que investi em cursos e leituras me fizeram ter oportunidades de conhecer muitas empresas, muitas pessoas, comunidades, trabalhos diferenciados e principalmente o aprendizado. 

Com o tempo, mudei minha forma de pensar e procurei seguir meu sonho. Me matriculei na FIAP, estou no meu segundo semestre e cara, posso ser sincero? Apesar do cansaço que as vezes nos consome, o sentimento de gratidão é enorme. Pois apesar de tudo, tenho absorvido tantos conceitos que as vezes eu não consigo assimilar rs. Sem contar o fato de que você _aprende do jeito certo_, uma vez que aprendendo sozinho - autodidata - você acaba criando vários _vícios_. (Há exceções? Não sei!)

Além disso, a FIAP abre portas, pois eu mesmo já comprovei. Fiz duas entrevistas indicadas pela FIAP e hoje trabalho em uma ótima empresa - falo da FIAP pois não sei se outras faculdades fazem o mesmo.

Portanto meu caro, existem 2 lados da moeda:
o autodidata - estuda por conta própria e rala pra conseguir algo legal.
o universitário - também estuda por conta própria rs, mas aprende com ótimos professores, rala - pq serio, não vou falar pra você que é fácil rs - e consegue oportunidades diferenciadas e únicas.

Bem, é isso.
Sucesso pra você, em sua escolha e lembre-se: _ninguém pode sonhar o seu sonho_!

**Comment by ghost** [comment]
Eu vi que a FIAP abriu o tecnologo em analise e desenvolvimento de sistemas online. Como é a experiência de curso Online? Alguém te experiência com a graduação Online e contando com a FIAP? Eu moro no interior de São Paulo, do lado de Campinas, se tiver alguém que  de campinas por aqui que saiba onde tenha algum tecnólogo presencial por aqui? Já cheguei a pesquisar e nada.  Se alguém tiver alguma experiência com esse curso https://www.fiap.com.br/online/graduacao/tecnologo/analise-e-desenvolvimento-de-sistemas/   

Vale a pena cursos online? A graduação de computação que eu estava fazendo era presencial, mas é um estresse. Cheguei a fazer um semestre na FATEC, mas acabei não me adaptando na fatec e achei a estrutura bem ruim, fora a rotina que eu tinha todo dia para ir, e tomava meu dia inteiro, então foi melhor trancar.   

Se alguém tiver experiência com curso Online principalmente o da FIAP, gostaria de saber como é. E se o tecnólogo online facilita da mesma maneira para arrumar um estágio.  

EDIT:  Cursos online no quesito de graduação mesmo. Sempre estudo por cursos livres pela UDEMY.

**Comment by vhfmag** [comment]
Muito obrigado pelo link, @simaodeveloper! Tava dando uma olhada em cursos online e o mais barato que eu tinha achado até então tinha sido o ['personalized learning' da Northern Arizona University](https://pl.nau.edu/), mas a University of the People tá muito mais em conta. A NAU é baseada em competências, então em tese dá pra terminar em menos tempo, mas acredito que minha carga de trabalho e falta de disciplina não permitiriam de qualquer maneira.

Uma pergunta: como você ficou sabendo dela? Conhece alguém que fez algum curso de lá? Quando eu tava pesquisando sobre essas universidades americanas com modalidade online, eu me perguntava muito se o diploma teria valor no mercado brasileiro ou não, nunca cheguei a achar alguém daqui que tivesse feito.

**Comment by ghost** [comment]
E @adhenrique  você conhece alguém que faz tecnólogo de ADS em ead pela FIAP? Eu vi pela FIAP e gostei bastante, creio que para mim, por mais que eu ainda não esteja no mercado de trabalho, estou a um ano tentando gostar do meio acadêmico e eu não consigo me adaptar e me desmotiva bastante de aprender. Gosto muito de ir por fora. Eu estudo mais fora da faculdade do que dentro da faculdade. Tenho feito uns cursos pela udemy e quero ver se perco logo o medo de criar meu portfolio e pegar projetos freelancer. Mas o tecnólogo ead da FIAP em ADS me agradou bastante, ainda mais das várias formas em que o conteúdo é passado. Eu só quero o diploma pois tenho interesse morar fora e facilita bastante. Se fosse pra ficar por aqui eu realmente não ligaria.

**Comment by adhenrique** [comment]
@brunojokubauskas eu não conheço, infelizmente. Mas já ouvi outras pessoas da sala falando de conhecidos que acham o EAD muito bom.

Na verdade, acredito que o único ponto a se levar em consideração na escolha de um curso EAD, é a auto-disciplina. Se você é disciplinado e tem força de vontade - como você mesmo falou - o EAD se encaixa perfeitamente para você. No meu caso eu ia me inscrever para o EAD, mas como sou um pouco indisciplinado acabei deixando a ideia de lado.

Quanto a pergunta _vale a pena fazer cursos online?_: Não tenha dúvidas disso. Mas lembre-se de pensar no ponto que citei acima.

**Comment by ghost** [comment]
Então, algo que eu percebi hoje, agora pouco. Estou levando mais a sério meus estudos no geral.  E eu percebi que eu tenho um certo medo de codar. Me sinto travado/perdido etc. Tenho feito um curso pela udemy e tenho estudado pelo menos 2h ou 4h por dia. Tentei reproduzir um twitter widget que vi no  Artigo do Zero ao Heroi Front End e me travei demais. Que dicas vocês podem me dar em relação a perder o medo  e para praticar? Quando estou praticando com o curso, sinto que flui e que entendo, mas quando vou para os experimentos me travo.

**Comment by mutheusalmeida** [comment]
@brunojokubauskas O mesmo acontece comigo. Eu acredito que seja por causa da ausência de um conteúdo real.

Uma boa dica quando estiver travando é tá sempre pesquisando as dúvidas que surgem e olhando o código de outros desenvolvedores.

**Comment by phhoffmann** [comment]
Fala @brunojokubauskas !

Cara, eu passo exatamente pela mesma coisa que você. Eu vejo os cursos e vou codando do lado e acho muito legal ver tudo funcionando. Na hora de sentar e fazer uma parada eu mesmo, sozinho, dá aquele bloqueio maluco e eu travo, parece que esqueci tudo o que eu fiz e saio correndo pros stack overflow e codepen da vida procurando o que eu preciso fazer.
Essa semana aqui no trampo rolou um trampo bem simples de um site one-page que ia ser passado pra um tercerizado fazer. Eu assumi a bronca, disse que parecia não ter nada complicado e que eu dava conta. 
Cara, sério.
**Que tesão.** 

Tô aprendendo flexbox numa loucura aqui todo dia (eu sou bem devagar no aprendizado eu acho hehe), tô curtindo muito. Só de conseguir fazer uma sessão ali se comportar do jeito que eu quero, a sensação de vitória é muito grande hahaha tá rolando uma liberação de endorfina com certeza!
Não sei se isso tem como acontecer na sua vida, de conseguir uns trampos mais reais pra treinar (eu to tendo essa liberdade aqui no trampo agora, antes nunca tive e vi vários trabalhos que eu talvez poderia ter feito sendo tercerizados), mas se você puder, tenta fazer isso.

Eu comecei a faculdade tem 2 semanas, tô com 30 anos e faz uns 8 que não pisava numa facul. Também to curtindo demais, se é necessário? Acho que não, tem muito auto didata aí falando que não é, né? Mas pra mim com certeza é. Depois de muitos cursos começados e parados no meio eu percebi que não é pra mim. Estar na facul, ter várias pessoas ali pra trocar ideia, me ajudando com o flexbox mesmo, eu tentando ajudar elas no que eu posso, a gente aprendendo junto...

Mano, tô descobrindo várias coisas massa que eu nem sabia que curtia! 
Enfim, se é necessário eu não sei. Mas tá sendo muito daora. (nesse momento eu acabei de conseguir fazer um negócio aqui e tô muito feliz então o texto talvez esteja "feliz demais")

Mas fica aí pra você saber que esses momentos existem, o desbloqueio existe. Mete a cara que dá certo, escreve esses código tudo ai, veio!

**Comment by dionarodrigues** [comment]
Vejam essa matéria que saiu no olhar digital sobre grandes empresas deixarem de exigir diploma.

https://olhardigital.com.br/pro/noticia/por-que-apple-e-google-nao-exigem-mais-diploma-universitario-para-contratar/78144

**Comment by mutheusalmeida** [comment]
O "Web Development Path" da Codecademy é uma boa escolha para se ter em um currículo de um Front-end Developer Junior?

**Comment by ninetails** [comment]
Só um fato que me ocorreu recentemente relacionado e gostaria de deixar um comentário raso: mesmo nossa área não requerer diploma, é sim algo a se mostrar na hora de pegar uma vaga em empresa grande e conceituada (mas um pouco antiquada).

**Comment by leandrocunha** [comment]
Depende, (como sempre). Na minha opnião, faculdade não chega a ser ponto crítico para adquirir conhecimento técnico, hoje em dia com muita dedicação vc consegue obter esse conhecimento na internet, com vivência no mercado e trocando com pessoas da área.

Pra conseguir empregos vc tem que botar na balança uma formação x tempo de experiência. Se vc tiver formação vc consegue uma vaga bacana ou se vc não tiver formação e tiver tempo de experiência vc tb consegue uma vaga bacana.

Agora se quiser pensar em conseguir algo fora do país a faculdade vale pontos extras na hora de te recrutarem, não pelo conhecimento técnico e sim como facilidade na hora de vc conseguir visto.

Estou vivendo essa experiencia agora, não tenho faculdade na área de tecnologia porem trabalho a 15 anos na área e olhando as vagas lá fora e conversando com recrutadores se eu tivesse uma faculdade eu já tinha conseguido sair do país a algum tempo por exemplo.

De qualquer maneira minha opnião é faça sim uma faculdade que só vai agregar conhecimento.

**Comment by samuelfaj** [comment]
Em minha humilde opinião, depende.

Depende muito dos seus objetivos profissionais e pessoais.
Para aprender a programar e obter um bom (para não dizer ótimo) conhecimento TÉCNICO a faculdade não vai lhe ajudar muito. Até porque nosso mundo é dinâmico demais, sai uma tecnologia nova agora, até que ela chegue a grade curricular já pode estar até defasada. 
Então para aprender a programar e evoluir tecnicamente diria que ser autodidata é essencial.

No meu caso, eu adoro programar, faço isso praticamente 24 horas por dia e sou bem autodidata.
Como profissão pretendo empreender, então, optei por largar o curso de analise e desenvolvimento de sistemas e estudar por conta, o que eu gosto e preciso para chegar ao meu objetivo.
Me arrependo? Não. Nem um pouco.

Enquanto meus amigos tem que perder tempo estudando matérias ultrapassadas e que eu julgo de pouca serventia, eu passo meu tempo estudando tecnologias atuais e fazendo projetos pessoais que me agregam muito valor.

Respeito a opinião dos colegas acima, apenas quis apresentar outro ponto de vista.

Sugiro também a pesquisa da seguinte notícia:
"Apple, Google e IBM já não exigem diploma universitário para contratar
Levantamento com empresas dos Estados Unidos mostra que muitas delas não tratam a formação acadêmica como requisito; entre elas estão Apple, Google e IBMApple, Google e IBM já não exigem diploma universitário para contratar"

**Comment by rubenmarcus** [comment]
alguem sabe dizer se os cursos da open university são bem vistos?
http://www.openuniversity.edu/iswcourses/programmes/subjects/computing-degrees
